source/extensions/internal_redirect/allowlisted_routes/allowlisted_routes.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "envoy/extensions/internal_redirect/allowlisted_routes/v3/allowlisted_routes_config.pb.h"
+#include "envoy/router/internal_redirect.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "absl/container/flat_hash_set.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace InternalRedirect {
+
+class AllowlistedRoutesPredicate : public Router::InternalRedirectPredicate {
+public:
+  AllowlistedRoutesPredicate(
+      absl::string_view,
+      const envoy::extensions::internal_redirect::allowlisted_routes::v3::AllowlistedRoutesConfig&
+          config)
+      : allowed_routes_(config.allowed_route_names().begin(), config.allowed_route_names().end()) {}
+
+  bool acceptTargetRoute(StreamInfo::FilterState&, absl::string_view route_name) override {
+    return allowed_routes_.contains(route_name);
```

Comment:
thoughts on case sensitivity?

Reply:
I don't have strong opinions here. Route.name() does not seem to be used much except by logging. So I think it may be good to be case insensitive? I can change the code that way and document the behavior in the API.